src/main/scala/tech/cryptonomic/conseil/tezos/TezosTypes.scala

```diff
@@ -5,15 +5,39 @@ package tech.cryptonomic.conseil.tezos
   */
 object TezosTypes {
 
+  //TODO use in a custom decoder for json strings that needs to have a proper encoding
+  lazy val isBase58Check: String => Boolean = (s: String) => {
+    val pattern = "^[123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz]*$".r.pattern
+    pattern.matcher(s).matches
+  }
+
   /** convenience alias to simplify declarations of block hash+level tuples */
   type BlockReference = (BlockHash, Int)
 
+  final case class PublicKey(value: String) extends AnyVal
+
+  final case class PublicKeyHash(value: String) extends AnyVal
+
+  final case class Signature(value: String) extends AnyVal
+
   final case class BlockHash(value: String) extends AnyVal
 
   final case class OperationHash(value: String) extends AnyVal
 
+  final case class ContractId(id: String) extends AnyVal
+
   final case class AccountId(id: String) extends AnyVal
 
+  final case class ChainId(id: String) extends AnyVal
+
+  final case class Nonce(value: String) extends AnyVal
+
+  final case class Secret(value: String) extends AnyVal
+
+  final case class MichelsonV1(expression: String) extends AnyVal
```

Comment:
Michelson is the normal representation of Michelson code. Micheline is the JSON representation. It seems we are storing the latter here.

Reply:
Great, I'll rename